netaddr.go

```diff
@@ -187,9 +149,7 @@ func ParseIP(s string) (IP, error) {
 			return IPv4(ip4[0], ip4[1], ip4[2], ip4[3]), nil
 		}
 	}
-	var a16 [16]byte
-	copy(a16[:], ipa.IP.To16())
-	return IPv6Raw(a16).WithZone(ipa.Zone), nil
+	return ipv6Slice(ipa.IP.To16()).WithZone(ipa.Zone), nil
```

Comment:
withInternedZone, found in https://github.com/inetaf/netaddr/pull/65, will make this more efficient.

Reply:
I don't follow how. In the current implementation of ParseIP, we're relying on the stdlib to give us the zone as a string. We have to intern it ourselves here, we don't yet have a *intern.Value. Am I missing something? (Other than the fact that ParseIP shouldn't be shelling out to net.ParseIP, but that's a known TODO)

Reply:
Oh, sorry, I got the type of `ipa` wrong. Pay me no mind. :)